rust/processor/src/processors/emojicoin_dot_fun/processor.rs

```diff
@@ -135,164 +183,696 @@ async fn insert_to_db(
+                                *melee_data = Some(MeleeData {
+                                    melee_id: melee.melee_id,
+                                    market_id_0: market_0.market_id,
+                                    market_id_1: market_1.market_id,
+                                    price_0: market_0.price,
+                                    price_1: market_1.price,
+                                });
+                            },
+                            ArenaEvent::Enter(enter) => {
+                                insert_events
+                                    .arena_position
+                                    .push(ArenaPositionDiffModel::from(enter.clone()));
+                                let model = ArenaEnterEventModel::new(txn_info.clone(), enter);
+                                insert_events
+                                    .arena_info_update
+                                    .push(ArenaInfoDiffUpdate::from(model.clone()));
+                                insert_events.arena_enter_events.push(model)
+                            },
+                            ArenaEvent::Exit(exit) => {
+                                insert_events
+                                    .arena_position
+                                    .push(ArenaPositionDiffModel::from(exit.clone()));
+                                let model = ArenaExitEventModel::new(
+                                    txn_info.clone(),
+                                    exit,
+                                    // This can never be None.
+                                    melee_data.as_ref().unwrap(),
+                                );
+                                insert_events
+                                    .arena_info_update
+                                    .push(ArenaInfoDiffUpdate::from(model.clone()));
+                                insert_events.arena_exit_events.push(model)
+                            },
+                            ArenaEvent::Swap(swap) => {
+                                ensure!(
+                                    last_two_states.0.is_some() && last_two_states.1.is_some(),
+                                    "The two previous State events related to this ArenaSwap event were not found."
+                                );
+                                let (mut state_0, mut state_1) = (None, None);
+                                std::mem::swap(&mut state_0, &mut last_two_states.0);
+                                std::mem::swap(&mut state_1, &mut last_two_states.1);
+                                let (state_0, state_1) = (state_0.unwrap(), state_1.unwrap());
+                                let (state_0, state_1) = if swap.emojicoin_0_proceeds.is_zero() {
+                                    (state_1, state_0)
+                                } else {
+                                    (state_0, state_1)
+                                };
```

Comment:
```suggestion
                                let state_a = last_two_states.0
                                    .take()
                                    .expect("The first state for this arena swap should be processed already.");
                                let state_b = last_two_states.1
                                    .take()
                                    .expect("The second state for this arena swap should be processed already.");

                                // Ensure the two state events are in the correct order.
                                let (state_0, state_1) = if swap.emojicoin_0_proceeds.is_zero() {
                                    (state_b, state_a)
                                } else {
                                    (state_a, state_b)
                                };
```

I believe this is more idiomatic?

Also, I'm a little confused about why emojicoin 0 proceeds means it's the second state event, and not the first..? What's the reasoning behind that? Just trying to understand